### PR TITLE
setup_requires to avoid break on bulk install

### PR DESCRIPTION
**Pull request by mabodo**
Solves issue #142

### CODE REVIEW

**Comment by mabodo** [comment]
The ultimate version do not use the setup_requires but changes the place from where to hardcode the version

File: setup.py [hunk 1]
```diff
@@ -3,11 +3,12 @@
 
 import os
 from setuptools import setup, find_packages
-from GPyOpt.__version__ import __version__
 
 def read(fname):
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
 
+__version__="1.2.1"
```
Comment (apaleyes) [inline setup.py]: Add some spaces: `__version__ = "1.2.1"`

File: GPyOpt/__version__.py [hunk 2]
```diff
@@ -1 +1,6 @@
-__version__ = "1.2.1"
+from pkg_resources import get_distribution, DistributionNotFound
```
Comment (apaleyes) [inline GPyOpt/__version__.py]: I am not familiar with this package, so a naive question maybe: is it always guaranteed to be available? Does it come with a standard Python installation, or perhaps we need to declare a new dependency for it?

Comment (mabodo) [reply GPyOpt/__version__.py]: almost standard :)

it comes with setuptools installation, which is not in the clean python installation, but most of python tutorials recommend you to install it.

I think you cannot use pip, nor setup.py without a setuptools installed :)

**Comment by GeoffDNeal** [comment]
Any chance this will end up in a new build on PyPI? Its pretty annoying to have install directly from github with pip install -r requirements

**Pull request merged**